src/transformers/generation/utils.py

```diff
@@ -2597,6 +2597,12 @@ def _contrastive_search(
         unfinished_sequences = torch.ones(batch_size, dtype=torch.long, device=input_ids.device)
         model_kwargs = self._get_initial_cache_position(input_ids, model_kwargs)
 
+        cosine_matrix_mask = None
+        if not self.config.is_encoder_decoder:
+            # Create cosine_matrix_mask based on the attention_mask
+            cosine_matrix_mask = model_kwargs["attention_mask"]
+            cosine_matrix_mask = cosine_matrix_mask.repeat_interleave(top_k, dim=0)
```

Comment:
Suggestion:

```suggestion
        attention_mask_name = "decoder_attention_mask" if self.config.is_encoder_decoder else "attention_mask"
        cosine_matrix_mask = model_kwargs[attention_mask_name]
        cosine_matrix_mask = cosine_matrix_mask.repeat_interleave(top_k, dim=0)
```

That way, your correction should also work in the case where the model is an encoder-decoder model and a partially masked decoder prompt is passed :)

(we may need extra code if `attention_mask_name` doesn't exist in `model_kwargs` or `model_kwargs[attention_mask_name]` can be `None`, let's see if the tests complain)